### PR TITLE
Bump to NLPModelsModifiers 0.7

### PR DESCRIPTION
**Pull request by tmigot**

### CODE REVIEW

**Comment by tmigot** [comment]
@dpo @amontoison It looks like we have a license issue here

**Comment by amontoison** [comment]
@tmigot
I use a link given by Oscar to download `libknitro.so` and `libiomp5.so` on his Google Drive.
Maybe the link is not working anymore.

@odow I recently downloaded all the versions of KNITRO 14 (new release) on the website of Artelys.
I propose to do an archive (*.zip) with all versions together and update the Julia script for CI in KNITRO.jl.
It will allow to test KNITRO.jl and NLPModelsKNITRO.jl on multiple platforms with only one secret environment.
Right we can only test it on Linux and we need to define an environment variable for the shared library of KNITRO and another one for the shared library of IntelOpenMP.

**Comment by dpo** [comment]
We’ll have an updated license on the M1 very soon.

**Comment by odow** [comment]
KNITRO.jl is still working https://github.com/jump-dev/KNITRO.jl/pull/283 so I'm not sure.

**Comment by odow** [comment]
If you could host binaries somewhere that would be useful. The drive link is a bit fragile because the file is 160 MB, and it trips Googles scanner.

**Comment by odow** [comment]
Here's an alternative: https://github.com/jump-dev/KNITRO.jl/pull/284

I can give you the link if you need. It's O(43 Mb) instead of 160.

**Comment by tmigot** [comment]
In the meantime, can somebody try this PR locally? Just to check that the test passes while using NLPModelsModifiers 0.7?

We should open an issue to fix the CI

**Comment by amontoison** [comment]
@tmigot 
I doesn't work withNLPModelsModifiers v0.6.6:
```julia
ERROR: LoadError: UndefVarError: `DiagonalQN` not defined
Stacktrace:
 [1] top-level scope
   @ ~/.julia/packages/NLPModelsModifiers/IMwDI/src/quasi-newton.jl:29
 [2] include(mod::Module, _path::String)
   @ Base ./Base.jl:495
 [3] include(x::String)
   @ NLPModelsModifiers ~/.julia/packages/NLPModelsModifiers/IMwDI/src/NLPModelsModifiers.jl:1
 [4] top-level scope
   @ ~/.julia/packages/NLPModelsModifiers/IMwDI/src/NLPModelsModifiers.jl:19
 [5] include
   @ ./Base.jl:495 [inlined]
 [6] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::String)
   @ Base ./loading.jl:2222
 [7] top-level scope
   @ stdin:3
in expression starting at /home/alexis/.julia/packages/NLPModelsModifiers/IMwDI/src/quasi-newton.jl:29
in expression starting at /home/alexis/.julia/packages/NLPModelsModifiers/IMwDI/src/NLPModelsModifiers.jl:1
in expression starting at stdin:3
ERROR: LoadError: Failed to precompile NLPModelsModifiers [e01155f1-5c6f-4375-a9d8-616dd036575f] to "/home/alexis/.julia/compiled/v1.10/NLPModelsModifiers/jl_tq83xE".
```

**Comment by tmigot** [comment]
OK, thanks. Does it work with 0.7?

**Comment by amontoison** [comment]
@tmigot 
```julia
(@v1.10) pkg> test NLPModelsKnitro
     Testing NLPModelsKnitro
      Status `/tmp/jl_ea1xqY/Project.toml`
  [54578032] ADNLPModels v0.7.0
  [67920dd8] KNITRO v0.14.1 `https://github.com/jump-dev/KNITRO.jl.git#master`
  [a4795742] NLPModels v0.20.0
  [bec4dd0d] NLPModelsKnitro v0.8.0 `~/Bureau/git/NLPModelsKnitro.jl`
  [e01155f1] NLPModelsModifiers v0.7.0
  [ff4d7338] SolverCore v0.3.7
  [37e2e46d] LinearAlgebra
  [8dfed614] Test
      Status `/tmp/jl_ea1xqY/Manifest.toml`
  [54578032] ADNLPModels v0.7.0
  [14f7f29c] AMD v0.5.3
  [6e4b80f9] BenchmarkTools v1.5.0
  [d360d2e6] ChainRulesCore v1.23.0
  [523fee87] CodecBzip2 v0.8.2
  [944b1d66] CodecZlib v0.7.4
  [ffa27691] ColPack v0.3.0
  [bbf7d656] CommonSubexpressions v0.3.0
  [34da2185] Compat v4.14.0
  [864edb3b] DataStructures v0.18.18
  [163ba53b] DiffResults v1.1.0
  [b552c78f] DiffRules v1.15.1
  [ffbed154] DocStringExtensions v0.9.3
  [e2ba6199] ExprTools v0.1.10
  [9aa1b823] FastClosures v0.3.2
  [f6369f11] ForwardDiff v0.10.36
  [069b7b12] FunctionWrappers v1.1.3
  [92d709cd] IrrationalConstants v0.2.2
  [692b3bcd] JLLWrappers v1.5.0
  [682c06a0] JSON v0.21.4
  [67920dd8] KNITRO v0.14.1 `https://github.com/jump-dev/KNITRO.jl.git#master`
  [40e66cde] LDLFactorizations v0.10.1
  [5c8ed15e] LinearOperators v2.7.0
  [2ab3a3ac] LogExpFunctions v0.3.27
  [1914dd2f] MacroTools v0.5.13
  [b8f27783] MathOptInterface v1.27.0
  [d8a4904e] MutableArithmetics v1.4.1
  [a4795742] NLPModels v0.20.0
  [bec4dd0d] NLPModelsKnitro v0.8.0 `~/Bureau/git/NLPModelsKnitro.jl`
  [e01155f1] NLPModelsModifiers v0.7.0
  [77ba4419] NaNMath v1.0.2
  [bac558e1] OrderedCollections v1.6.3
  [69de0a69] Parsers v2.8.1
  [aea7be01] PrecompileTools v1.2.1
  [21216c6a] Preferences v1.4.3
  [ae029012] Requires v1.3.0
  [37e2e3b7] ReverseDiff v1.15.1
  [ff4d7338] SolverCore v0.3.7
  [276daf66] SpecialFunctions v2.3.1
  [90137ffa] StaticArrays v1.9.3
  [1e83bf80] StaticArraysCore v1.4.2
  [a759f4b9] TimerOutputs v0.5.23
⌃ [3bb67fe8] TranscodingStreams v0.9.13
  [6e34b625] Bzip2_jll v1.0.8+1
  [f218ff0c] ColPack_jll v0.3.0+0
  [efe28fd5] OpenSpecFun_jll v0.5.5+0
  [0dad84c5] ArgTools v1.1.1
  [56f22d72] Artifacts
  [2a0f44e3] Base64
  [ade2ca70] Dates
  [f43a241f] Downloads v1.6.0
  [7b1f6079] FileWatching
  [b77e0a4c] InteractiveUtils
  [b27032c2] LibCURL v0.6.4
  [76f85450] LibGit2
  [8f399da3] Libdl
  [37e2e46d] LinearAlgebra
  [56ddb016] Logging
  [d6f4376e] Markdown
  [a63ad114] Mmap
  [ca575930] NetworkOptions v1.2.0
  [44cfe95a] Pkg v1.10.0
  [de0858da] Printf
  [9abbd945] Profile
  [3fa0cd96] REPL
  [9a3f8284] Random
  [ea8e919c] SHA v0.7.0
  [9e88b42a] Serialization
  [6462fe0b] Sockets
  [2f01184e] SparseArrays v1.10.0
  [10745b16] Statistics v1.10.0
  [fa267f1f] TOML v1.0.3
  [a4e569a6] Tar v1.10.0
  [8dfed614] Test
  [cf7118a7] UUIDs
  [4ec0a83e] Unicode
  [e66e0078] CompilerSupportLibraries_jll v1.1.0+0
  [deac9b47] LibCURL_jll v8.4.0+0
  [e37daf67] LibGit2_jll v1.6.4+0
  [29816b5a] LibSSH2_jll v1.11.0+1
  [c8ffd9c3] MbedTLS_jll v2.28.2+1
  [14a3606d] MozillaCACerts_jll v2023.1.10
  [4536629a] OpenBLAS_jll v0.3.23+4
  [05823500] OpenLibm_jll v0.8.1+2
  [bea87d4a] SuiteSparse_jll v7.2.1+1
  [83775a58] Zlib_jll v1.2.13+1
  [8e850b90] libblastrampoline_jll v5.8.0+1
  [8e850ede] nghttp2_jll v1.52.0+1
  [3f19e933] p7zip_jll v17.4.0+2
        Info Packages marked with ⌃ have new versions available and may be upgradable.
     Testing Running tests...
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
┌ Warning: Knitro only treats bound-constrained least-squares problems; converting to feasibility form
└ @ NLPModelsKnitro ~/Bureau/git/NLPModelsKnitro.jl/src/api.jl:105
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
##### This license is only intended for use by POLYTECH. #####
##### License is valid until oct 19, 2024 #####
     Testing NLPModelsKnitro tests passed
```

File: Project.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ ADNLPModels = "0.7"
 KNITRO = "0.14"
 LinearAlgebra = "1.6"
 NLPModels = "0.18, 0.19, 0.20"
-NLPModelsModifiers = "0.6"
+NLPModelsModifiers = "0.6, 0.7"
```
Comment (amontoison) [inline Project.toml:16]:
```suggestion
NLPModelsModifiers = "0.7"
```

**Review by amontoison — APPROVED** [review]

**Pull request merged**